src/charm.py

```diff
@@ -49,6 +49,27 @@ def _config_root_ca_certificate_validity(self) -> int:
         """
         return int(self.model.config.get("root-ca-validity"))  # type: ignore[arg-type]
 
+        self.framework.observe(
+            self.on.get_issued_certificates_action, self._on_get_issued_certificates
+        )
+
+    def _on_get_issued_certificates(self, event: ActionEvent) -> None:
+        """Handler for the get-issued-certificates action.
+
+        Outputs the issued certificates per application.
+
+        Args:
+            event (ActionEvent): Juju event.
+
+        Returns:
+            event (ActionEvent): Juju event.
+        """
+        certificates = self.tls_certificates.get_issued_certificates()
+        if not certificates:
+            event.fail("No certificates issued yet.")
+            return
+        event.set_results({"Issued": certificates})
```

Comment:
I think all our keys are in lower case, and I would make it a bit more explicit.

```suggestion
        event.set_results({"issued_certificates": certificates})
```

Reply:
Fixed